Packs/CommonScripts/Scripts/SetGridField/SetGridField.py

```diff
@@ -110,23 +181,31 @@ def build_grid(context_path: str, keys: List[str], columns: List[str]) -> pd.Dat
         context_path: DT context path.
         keys: Keys to be included
         columns: Grid columns name.
+        skip_nested_elements: False for unpacking nested elements, True otherwise.
 
     Returns:
         pd.DataFrame: New Table include data from Entry Context
     """
     # Retrieve entry context data
     entry_context_data = demisto.dt(demisto.context(), context_path)
     # Validate entry context structure
-    validate_entry_context(entry_context_data, keys)
+    data_type = validate_entry_context(entry_context_data, keys, skip_nested_elements)
     # Building new Grid
-    if len(entry_context_data) > 1:
-        # Handle entry context list option
+    if not skip_nested_elements:
+        table = pd.DataFrame(unpack_all_data_from_dict(entry_context_data, keys, columns))
+        table.rename(columns=dict(zip(table.columns, columns)), inplace=True)
+    elif data_type == 'list':
+        # Handle entry context as list of value
+        table = pd.DataFrame(entry_context_data)
+        table.rename(columns=dict(zip(table.columns, columns)), inplace=True)
```

Comment:
don't you want to unpack here too? or is it not the usecase?

Reply:
This is not the use-case here, this is intended to a dict with primitive types in the values only.
Updated the comment.